File: app/src/main/java/com/musicplayer/SocyMusic/data/Playlist.java
package com.musicplayer.SocyMusic.data;

import android.content.Context;
import android.media.MediaMetadataRetriever;
import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.UUID;

@Entity
public class Playlist implements Serializable {
    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "playlist_id")
    private final UUID id;
    @ColumnInfo(name = "playlist_name")
    private String name;
    @Ignore
    private ArrayList<Song> songList;

    public Playlist(@NonNull UUID id, String name, ArrayList<Song> songList) {
        this.id = id;
        this.name = name;
        this.songList = songList;
    }

    public Playlist(@NonNull UUID id, String name) {
        this.id = id;
        this.name = name;
        songList = new ArrayList<>();
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getSongCount() {
        return songList.size();
    }

    public ArrayList<Song> getSongList() {
        return songList;
    }

    public void setSongList(ArrayList<Song> songList) {
        this.songList = songList;
    }

    @NonNull
    public UUID getId() {
        return id;
    }

    public boolean isFavorites() {
        return id.equals(SongsData.FAVORITES_PLAYLIST_ID);
    }

    public boolean contains(Song song) {
        return songList.contains(song);
    }

    public void addSong(Song song) {
        songList.add(song);
    }

    public void removeSong(Song song) {
        songList.remove(song);
    }

    public int calculateTotalDuration() {
        MediaMetadataRetriever metadataRetriever = new MediaMetadataRetriever();
        int totalDuration = 0;
        for (Song song : songList) {
            metadataRetriever.setDataSource(song.getPath());
            totalDuration += Integer.parseInt(metadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION));
        }
        return totalDuration;
    }

}

File: app/src/main/kotlin/com/musicplayer/openmusic/data/Playlist.kt
package com.musicplayer.openmusic.data

import android.media.MediaMetadataRetriever
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity
class Playlist : Serializable {
    @PrimaryKey
    @ColumnInfo(name = "playlist_id")
    val id: UUID

    @ColumnInfo(name = "playlist_name")
    var name: String

    @Ignore
    var songList: MutableList<Song>

    constructor(id: UUID, name: String, songList: MutableList<Song>) {
        this.id = id
        this.name = name
        this.songList = songList
    }

    constructor(id: UUID, name: String) {
        this.id = id
        this.name = name
        songList = mutableListOf()
    }

    val songCount: Int
        get() = songList.size

    fun getSongAt(index: Int): Song {
        return songList[index]
    }

    val isFavorites: Boolean
        get() = id == SongsData.FAVORITES_PLAYLIST_ID

    operator fun contains(song: Song): Boolean {
        return songList.contains(song)
    }

    fun addSong(song: Song) {
        songList.add(song)
    }

    fun removeSong(song: Song) {
        songList.remove(song)
    }

    fun removeSongAt(index: Int) {
        songList.removeAt(index)
    }

    fun calculateTotalDuration(): Int {
        val metadataRetriever = MediaMetadataRetriever()
        var totalDuration = 0
        for (song in songList) {
            metadataRetriever.setDataSource(song.path)
            totalDuration += metadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)!!
                .toInt()
        }
        return totalDuration
    }

    override fun equals(other: Any?): Boolean {
        return if (other !is Playlist) false else other.id == id
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + name.hashCode()
        result = 31 * result + songList.hashCode()
        result = 31 * result + songCount
        result = 31 * result + isFavorites.hashCode()
        return result
    }
}